tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+        project = factories.ProjectFactory()
+        factories.URLFactory(url_project=project, attribute='surt')
+        request = rf.post('/', {'partial-search': ''})
+
+        with pytest.raises(Http404):
+            views.url_lookup(request, project.project_slug)
+
+    def test_template_used(self, client):
+        project = factories.ProjectFactory()
+        url = factories.URLFactory(url_project=project, attribute='surt')
+        response = client.post(
+            reverse('url_lookup', args=[project.project_slug]),
+            {'search-url-value': url.entity, 'partial-search': ''}
+        )
+
+        assert response.templates[0].name == 'nomination/url_search_results.html'
+
+    def test_context(self, client):
+        project = factories.ProjectFactory()
+        url = factories.URLFactory(url_project=project, attribute='surt')
+        response = client.post(
+            reverse('url_lookup', args=[project.project_slug]),
+            {'search-url-value': url.entity, 'partial-search': ''}
+        )
+
+        assert response.context['project'] == project
+        assert len(response.context['url_list']) == 1
+        assert response.context['url_list'][0] == url
+
+
+class TestProjectUrls():
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_status_ok(self, client, request_type):
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_template_used(self, client, request_type):
+        project = factories.ProjectFactory()
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+        assert response.templates[0].name == 'nomination/project_urls.html'
+
+    @pytest.mark.parametrize('request_type', [
+        'get',
+        'post'
+    ])
+    def test_context(self, client, request_type):
+        project = factories.ProjectFactory()
+        factories.URLFactory.create_batch(
+            3,
+            url_project=project,
+            attribute='surt',
+            value='http://(com,example.www)'
+        )
+        factories.NominatedURLFactory.create_batch(
+            2,
+            url_project=project,
+            value=1
+        )
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+
+        assert response.context['project'] == project
+        assert response.context['url_count'] == 3
+        assert response.context['nominator_count'] == 2
+        for key, value in response.context['browse_tup']:
+            assert key == 'com'
+            for each in value:
+                if each[1] is not None:
+                    assert each == ('E', 'http://(com,e')
```

Comment:
Yeah, you are correct. Do you think I should just cut out the 'for' loop altogether?